### DISCUSSION TITLE
Working with cucumber-rust projects in CLion

### DISCUSSION THREAD

**Discussion by thomasmathews01** [cucumber-rs/cucumber#218] [Q&A] [answered]
Is there a nice way of using standard IDEs with cucumber-rust projects? 

I typically use CLion for Rust, which has no way of switching off the extra command line params it sends with the 'cargo test' command to allow it to parse the normal unit test results for display in the GUI. These extra params cause the cucumber framework to crash out (`cargo test --color=always --no-fail-fast -- --format=json -Z unstable-options --show-output`), and as such you can't use the IDE-bundled debugger.

It is useable by running entirely from the terminal with 'cargo test', and then using println! statements for developing against, but this is not ideal and prevents you from setting breakpoints (and with async code sometimes it seems to exit before the println gets printed on my machine if it panics, which is very unhelpful).

Does anyone know if there is a way I can get the BDD to run under 'cargo run' instead, which will have no extra arguments and then hopefully allow debugging etc through the normal IDE functions?

And if not what approach is everyone else taking to develop when using this library?

**Comment by tyranron** [comment, marked as answer]
@thomasmathews01 see https://github.com/cucumber-rs/cucumber/discussions/203 and so https://github.com/cucumber-rs/cucumber/issues/194
